### DISCUSSION TITLE
Chunked uploads with Google Cloud Service & 308 responses

### DISCUSSION THREAD

**Discussion by pete-moody** [rpldy/react-uploady#342] [Q&A] [answered]
Is there a way to get the chunked upload to resume on a 308 response instead of a 200 response?

See [https://cloud.google.com/storage/docs/performing-resumable-uploads](https://cloud.google.com/storage/docs/performing-resumable-uploads)

For each chunk uploaded,
"If the request succeeds, the server responds with 308 Resume Incomplete. The response contains a [Range header](https://cloud.google.com/storage/docs/json_api/v1/parameters#range). Use the upper value in this header to determine where to start the next chunk. You should not assume that the server received all bytes sent in the request."

"Once the upload completes in its entirety, you receive a 200 OK or 201 Created response, along with any metadata associated with the resource." - i.e. GCS only responds with a 200 once the last chunk has been successfully received.

So I think I need to intercept somehow the upload response "error", check for a 308, then allow it to carry on. Is this possible? Or do I need to create a brand new enhancer for GCS chunked uploads?

**Comment by yoavniran** [comment]
Hey @pete-moody 
Looks like its currently not going to be easy to provide an override.
I think the best short term solution would be for me to add a custom success status codes prop that you will be able to provide. This will be a good enhancement in general and will allow you to pass `[308]` and it will be considered success for your uploads. I will try and make this available in the next few days

In the longer term I would definitely like to create a proper gcs sender for Uploady. I will need to learn more about their protocol around resumable uploads.

**Reply by pete-moody** [reply]
Thanks @yoavniran . A custom success code prop should work for us.

**Comment by yoavniran** [comment, marked as answer]
@pete-moody v0.18.0 is now out with a new sender option: [isSuccessfulCall](https://react-uploady.netlify.app/docs/api/#issuccessfulcall)
You can implement the callback to check the status of the request and return true in case of 308. 
Let me know if you have questions or issues.

**Reply by pete-moody** [reply]
Thanks! This works well as a solution for GCS chunked uploads
